pywbem_mock/_baserepository.py

```diff
@@ -0,0 +1,395 @@
+          the returned object does not change the data store. The default
+          behavior is True .  If copy is False, the object in the object store
+          is returned but if it is modified by the user, the object in the
+          store may be modified also.
+
+        Returns:
+
+            Returns the cim_object identified by the name parameter
+
+        Raises:
+
+            KeyError: cim_object with name not in this object store
+        """
+        pass
+
+    @abstractmethod
+    def create(self, name, cim_object):
+        """
+        Adds the cim_object identified by name to the object store.
+
+        Parameters:
+
+          name(:term:`string` or :class:`~pywbem.CIMInstanceName`):
+            Name by which the object is identified in the object store .
+
+          cim_object(:term:`CIM object` to be added to the object store):
+            The CIM object that will be inserted into the object store. The
+            object is copied into the object store so the user can safely
+            modify the original object without affecting the store.
+
+        Raises:
+
+            ValueError: If object already exists in the object store.
+        """
+        pass
+
+    @abstractmethod
+    def update(self, name, cim_object):
+        """
+        Replace the CIM object defined by name in the object store with
+        the object defined by cim_object.
+
+        Parameters:
+
+          name(:term:`string` or :class:`~pywbem.CIMInstanceName`):
+            Name by which the object is identified in the object store.
+
+          cim_object(:term:`CIM object`):
+            The CIM object to replace the original CIM object in the data
+            store. The object is copied into the object store so the user can
+            safely modify the original object without affecting the store.
+
+        Raises:
+
+            KeyError: If no object with name exists in the object store.
+        """
+        pass
+
+    @abstractmethod
+    def delete(self, name):
+        """
+        Delete the CIM object identified by name from the object store.
+
+        Parameters:
+
+          name(:term:`string` or :class:`~pywbem.CIMInstanceName`):
+            Name by which the object to be deleted is identified in the object
+            store.
+
+        Raises:
+
+            KeyError: If there is no object with name in this object store
+        """
+        pass
+
+    @abstractmethod
+    def iter_names(self):
+        """
+        Return iterator to the names of the CIM objects in the object store.
+        The order of returned names is undefined. Objects may be accessed using
+        iterator methods. The order of returned object is undetermined.
+
+        Returns:
+
+            Python iterator for the names of CIM objects in the object store.
+        """
+        pass
+
+    @abstractmethod
+    def iter_values(self, copy=True):
+        """
+        Return iterator to the cim objects in the object store. This allows
```

Comment:
Change "cim objects" to "CIM objects".

Reply:
DONE. Fixed lower case CIM in two locations.